shells/teepot/default.nix

```diff
@@ -7,18 +7,36 @@
 , taplo
 , vault
 , cargo-release
+, nixsgx
+, stdenv
 }:
 mkShell {
   inputsFrom = [ teepot.teepot ];
 
-  shellHook = ''
-    export OPENSSL_NO_VENDOR="1";
-  '';
-
   packages = [
     dive
     taplo
     vault
     cargo-release
   ];
+
+  TEE_LD_LIBRARY_PATH = lib.makeLibraryPath [
+    nixsgx.sgx-dcap
+    nixsgx.sgx-dcap.quote_verify
+    nixsgx.sgx-dcap.default_qpl
+  ];
+
+  QCNL_CONF_PATH = "${nixsgx.sgx-dcap.default_qpl}/etc/sgx_default_qcnl.conf";
+  OPENSSL_NO_VENDOR = "1";
+
+  shellHook = ''
+    if [ "x$NIX_LD" = "x" ]; then
```

Comment:
Interesting. I didn’t know the `x` prefix [was/is a common defensive practice](https://unix.stackexchange.com/q/32180).